### PR TITLE
Chemical foam dart

### PR DESCRIPTION
**Pull request by shazbot194**
image

This PR adds in foam darts that work just like syringe guns, but have a max cappacity of 5u. They can be made at a hacked autolathe for 1500 metal, which is 1.5x the amount a riot dart costs, .375x the cost of a shotgun dart, which can hold 30 units, and 150x the metal cost of a syringe which costs 10 metal. These do feed into any foam force gun, however, they are not re-useable.

🆑 Shazbot
Adds: Adds in Foam Dart Syringes and lets them be made at a hacked autolathe.
/🆑

### CODE REVIEW

**Comment by DesolateG** [comment]
Do reagents proc mixture upon impact?

**Comment by shazbot194** [comment]
It should mix inside the dart

**Comment by alexkar598** [comment]
5u is enough to kill with heparin,scichem mains(yes that exists) will go and order foam gun at cargo and make then self an unsuspectable,concealed weapon with a 1 shot hit to kill

**Comment by Spartan6** [comment]
Should be restricted to traitor ammo boxes.

**Comment by shazbot194** [comment]
While 5u might be enough to kill, it's a rather slow death, or at least should be, not to mention you can use something like a shotgun dart for 30u, which is enough to kill very fast. Which is also why I don't think these should be a traitor only item, as while they are good for what they are, they are not the end all be all for this type of item. Not to mention I didn't make a box for them, just singles.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 1]
```diff
@@ -270,6 +270,22 @@
 	projectile_type = /obj/item/projectile/bullet/dart/syringe/tranquilizer
 	materials = list(MAT_METAL=250)
 
+/obj/item/ammo_casing/foamsyringe
+	name = "foam dart syringe"
+	desc = "A foam dart with a small syringe in it. It looks to fragile to be re-used."
```
Comment (Birdtalon) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: `too fragile`

**Comment by FalseIncarnate** [comment]
Not sure i like the idea of what is effectively a stealth syringe gun, even if the shots are smaller. This really would only be useful for science (who can make real syringe guns) or cargo traitors, so I'm not sure what the real intent of this is.

The plus side is this still respects THICKMATERIAL, so at least hardsuits and bio suits protect against these

**Comment by shazbot194** [comment]
Well, this would still need a chemical to use, and its a lot louder, even if it's search stealthy. So realistically, the people who would use this would be people who have access to chems, or who can get access to chems, such as an officer who wants to use Haloperidol or a medic who wants to use Epinephrine/Atropine at small ranges, and where a small amount is more then enough. This is the major use I foresee, along with more nefarious tasks.

**Comment by FalseIncarnate** [comment]
Almost every case you gave us already possible. Sec has haloperidol darts in the armory, epinephrine is in the auto injectors literally everyone gets, and cures can be put in a syringe gun which the CMO can easily retrieve from storage. All of these cases don’t require a hacked autolathe and are essentially available at roundstart.

**Comment by shazbot194** [comment]
There is a major difference however. While yes, all are more or less already possible, there are normally massive limits on how you currently can. While haloperidol darts are in the armory, almost no-one knows how to use them, not to mention you need a shotgun to be able to use them, which I am willing to bet 9 times out of 10, anyone who gets a shotgun would just load it with buckshot anyway unless the meth head problem was assuredly massive and they were more or less forced to use them. While Auto-injectors are avalible to paramedics, they both take up a fair amount of space, and are single use, and 15u may not be needed for everyone, and in some cases, may even be deadly if someone already injected them before. Not to mention this is a lot slower when you have a lot of people who need help ASAP who are not next to eachother, such as after a fight. And as for the CMO with a syringe gun, I admit that ya, that does work out rather well, but you can give people a lot more smaller doses with something like the foam pistol over the syringe gun. I will admit that these are rather niche, but I think they are a great contribution, and if you want, I can go back and make all the foam magazines show what the top dart is at a visual glance.

**Comment by AdamElTablawy** [comment]
you underestimate how powerful 5u of a given chem mix can be, and also how many times people will shoot someone with their meme foam pistol.

one 5u mix is enough to kill someone if you know what you're doing. as it is, this is basically a rapid syringe gun, except with a smaller sprite and a somewhat smaller dosage.

also, with sufficient effort, someone could get their hands on a donksoft C-20 or LMG and then you have hell on earth if they decide to replace the rounds with the darts. an LMG syringe gun is going to be funny only the first time.

**Comment by shazbot194** [comment]
> also, with sufficient effort, someone could get their hands on a donksoft C-20 or LMG and then you have hell on earth if they decide to replace the rounds with the darts. an LMG syringe gun is going to be funny only the first time.

Currently there is no way to get a donksoft gun outside of the pistol or through nukies. You can get one from admins, but I do doubt they would give one out for use with chemdarts. 

> you underestimate how powerful 5u of a given chem mix can be, and also how many times people will shoot someone with their meme foam pistol.
one 5u mix is enough to kill someone if you know what you're doing. as it is, this is basically a rapid syringe gun, except with a smaller sprite and a somewhat smaller dosage.

The main problem with this, is that in order to be able to make these chem mixes, you either need to be a chemist or a scientist, with a scientist already normally being able to make a rapid syring gun. There are some other ways to get deadly chems for this, but they are not nearly as deadly, for the most part, and would have a harder time being made into perfect 5u mixes. I would also like to point out the shotgun dart which can hold 30u and can be loaded into a sawn off improvised shotgun, which if it hits, would be the same as shooting someone with 6 of these darts, for a fraction of the material cost.

**Comment by shazbot194** [comment]
Thanks @Squirgenheimer , you were a massive help in fixing this PR, and all future PRs.

**Comment by shazbot194** [comment]
Oh, oops, was on the wrong branch, didn't need to update this one

**Comment by DarkPyrolord** [comment]
This one is going to be a close.

The issue has been bought up that people do have trouble with syringes guns as is, especially with how deadly they can be as is, this is pretty much giving foam weapons the ability to act as an RSG while not looking like an RSG, which is a huge bonus for traitors and such with minimal risk before firing it off.

**Pull request closed without merging**